### PR TITLE
can now read or write to general IO rather than just the filesystem

### LINKED ISSUE #113: make it possible to readwrite from any `IO` object, not just files
Not sure why in the world I was stupid enough not to do this in the first place.  Very useful for, for example, writing to S3 buckets or other IO that doeshn't (or ideally wouldn't) directly involve the file system.

I should be able to get to it soon.

### PR DESCRIPTION
**Pull request by ExpandingMan**
Fixes #113.  This is especially useful for doing things like reading and writing to Amazon S3 or anything else over a network (previously one would have had to write to disk first).

### CODE REVIEW

**Comment by ExpandingMan** [comment]
@quinnj , tests are failing because in the docker container Python segfaults when reading `test2.feather` in the docker container.  However, I read the same file in Python locally and it works perfectly fine.  Is it possible that the Python `feather` package or something else in the container needs updating?  I can reproduce the segfault within the container, but only within the container.  Anywhere else I read the `test2.feather` file in either Julia or Python it works perfectly fine.

**Comment by ExpandingMan** [comment]
Much weirder: the test passes with no issues on Julia nightly on both linux and mac.  That makes absolutely no sense to me whatsoever.  Any idea what might be going on with that container?

**Comment by ExpandingMan** [comment]
This package is basically unusable for me now because it doesn't have this.

Anybody have any ideas what's going on in the docker?  Is there anyway to trigger this again?  I'm totally and completely stumped I was never able to reproduce the error anywhere.  Perhaps I should check if it still occurs on Julia 1.1...

**Comment by ExpandingMan** [comment]
I'm in the process of updating this and also implementing better (but optional) round-trip python tests as discussed in #119.

**Comment by ExpandingMan** [comment]
Ok, I have removed all of the docker stuff completely.  There is now an optional set of Python round-trip tests (the same as before).  They can't run in travis because it will be missing dependencies, but anyone should have no trouble running them locally, one just needs `pandas`, and `pyarrow`.  In the future if anyone can find a way to provision travis with this, those tests can be restored.

Note that the tests involving the Python byte array type have been removed.  They *were not working in master either*.  The reason this went unnoticed is because the error only seems to occur with updated `pandas` and `pyarrow`.  We can fix this later (I still fully intend on finishing radically overhaulingg Arrow.jl, sorry I got stalled out for a while).

This should also work with the latest DataFrames.jl.  I'd like to make another PR later to remove the DataFrames dependence entirely in favor of the Tables.jl interface only, but of course these will be breaking changes.

If anyone has any issues, please say so, otherwise I will merge this tomorrow.

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+[deps]
+Arrow = "69666777-d1a9-59fb-9406-91d4454c9d45"
+CategoricalArrays = "324d7699-5711-5eae-9e2f-1d82baa6b597"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
+FlatBuffers = "53afe959-3a16-52fa-a8da-cf864710bae9"
+Mmap = "a63ad114-7e13-5084-954f-fe012c677804"
+Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
+
```
Comment (quinnj) [inline Project.toml]: It'd be good to have a compat section here too; at least for julia, but probably also for DataFrames (it just released a breaking release in 0.18)

Comment (ExpandingMan) [reply Project.toml]: Done for julia and DataFrames.

**Comment by ExpandingMan** [comment]
I'm going to merge this today.  Ideally we should tag this today as well, any objections?

**Comment by ExpandingMan** [comment]
Does anyone know if we have to do anything new with Registrator, or can we just tag?

**Comment by sglyon** [comment]
I think tagging will not do anything.

I've had success:

- edit Project.toml to update version number
- Go to commit we want to tag version for (e.g. https://github.com/JuliaData/Feather.jl/commit/2153ec354de6a463e158af8bfb7124407278deda)
- leave `@JuliaRegistrator register()` as a comment on the commit

**Comment by ExpandingMan** [comment]
I took the liberty of bumping the version number to `0.5.2` in `Project.toml` by committing directly to master.

@quinnj you may need to call the registrator as I am not a member of JuliaData.

Is this seriously completely decoupled from tagging now?  I don't think I like that.

**Comment by KristofferC** [comment]
> Is this seriously completely decoupled from tagging now? I don't think I like that.

You can install tagbot that will tag automatically for you or you can paste the git command that Registrator gives you to tag yourself.

**Comment by ExpandingMan** [comment]
That seems reasonable.

Still seems that Quinn needs to do this.

**Comment by quinnj** [comment]
I kicked of JuliaRegistrator; I also made @ExpandingMan an admin on this repo; is that enough to allow him to call JuliaRegistrator? Anybody know?

**Comment by ExpandingMan** [comment]
According to the message that was given to @sglyon , I would need to be a member of JuliaData.

**Comment by sglyon** [comment]
I guess you’ll just have to develop more awesome features quickly so you can tag again and test it out ;)

**Pull request merged**